src/main/java/frc/robot/subsystems/vision/VisionIOPhotonLib.java

```diff
@@ -0,0 +1,197 @@
+        _poseEstimator.setMultiTagFallbackStrategy(PoseStrategy.LOWEST_AMBIGUITY);
+
+        NetworkTableInstance.getDefault().addListener(NetworkTableInstance.getDefault().getEntry("/photonvision/" + Constants.Vision.PHOTON_CAMERA_NAME + "/latencyMillis"), EnumSet.of(NetworkTableEvent.Kind.kValueRemote), event ->
+        {
+            var results = _camera.getAllUnreadResults();
+            if (results.isEmpty())
+            {
+                return;
+            }
+            // get the latest result (last in the list)
+            var result = results.get(results.size() - 1);
+
+            double                    timestamp = result.getTimestampSeconds();
+            List<PhotonTrackedTarget> targets   = result.getTargets();
+
+            List<Double>              distances        = new ArrayList<>();
+            List<Integer>             ids              = new ArrayList<>();
+            List<Double>              yaws             = new ArrayList<>();
+            List<PhotonTrackedTarget> processedTargets = new ArrayList<>();
+
+            // Process all targets
+            for (PhotonTrackedTarget target : targets)
+            {
+                ids.add(target.getFiducialId());
+                yaws.add(target.getYaw());
+
+                var transform = target.getBestCameraToTarget();
+                var distance  = Math.hypot(transform.getX(), transform.getY());
+                distances.add(distance);
+
+                if (distance <= Constants.Vision.MAX_DETECTION_RANGE)
+                {
+                    processedTargets.add(target);
+                }
+            }
+
+            // Get pose estimate from processed targets
+            _poseEstimator.setReferencePose(drive.getPose());
+            Optional<EstimatedRobotPose> estimatedPose = _poseEstimator.update(result);
+            Pose2d                       pose          = new Pose2d();
+            boolean                      hasPose       = false;
+
+            if (estimatedPose.isPresent())
+            {
+                pose    = estimatedPose.get().estimatedPose.toPose2d();
+                hasPose = true;
+            }
+
+            synchronized (VisionIOPhotonLib.this)
+            {
+                _captureTimesStamp   = timestamp;
+                _distances           = distances.stream().mapToDouble(Double::doubleValue).toArray();
+                _pose                = pose;
+                _numProcessedTargets = processedTargets.size();
+                _hasPose             = hasPose;
+                _targetIds           = ids.stream().mapToInt(Integer::intValue).toArray();
+                _targetYaws          = yaws.stream().mapToDouble(Double::doubleValue).toArray();
+            }
+        });
+    }
+
+    public Optional<EstimatedRobotPose> getEstimatedGlobalPose(Pose2d prevEstimatedRobotPose, PhotonPipelineResult result)
```

Comment:
This method is only visible in the `VisionIOPhotonLib` class, and although we create an instance of this in `RobotContainer`, we'll refer to it as a `VisionIO` object at that point meaning we won't have visibility to this method. Because of this, I'd like this method to be private.

```suggestion
    private Optional<EstimatedRobotPose> getEstimatedGlobalPose(Pose2d prevEstimatedRobotPose, PhotonPipelineResult result)
```